crates/bevy_pbr/src/meshlet/cull_meshlets.wgsl

```diff
@@ -48,61 +73,60 @@ fn cull_meshlets(@builtin(global_invocation_id) cluster_id: vec3<u32>) {
+    culling_bounding_sphere_radius = previous_model_scale * bounding_spheres.self_culling.radius;
 #else
-    var meshlet_visible = get_meshlet_previous_occlusion(cluster_id.x);
-    if !meshlet_visible { return; }
+    let culling_bounding_sphere_center_view_space = (view.inverse_view * vec4(culling_bounding_sphere_center.xyz, 1.0)).xyz;
 #endif
 
-    // Calculate world-space culling bounding sphere for the cluster
-    let culling_bounding_sphere_center = model * vec4(bounding_spheres.self_culling.center, 1.0);
-    let culling_bounding_sphere_radius = model_scale * bounding_spheres.self_culling.radius;
-
-    // Frustum culling
-    // TODO: Faster method from https://vkguide.dev/docs/gpudriven/compute_culling/#frustum-culling-function
-    for (var i = 0u; i < 6u; i++) {
-        if !meshlet_visible { break; }
-        meshlet_visible &= dot(view.frustum[i], culling_bounding_sphere_center) > -culling_bounding_sphere_radius;
+    let aabb = project_view_space_sphere_to_screen_space_aabb(culling_bounding_sphere_center_view_space, culling_bounding_sphere_radius);
+    let depth_pyramid_size_mip_0 = vec2<f32>(textureDimensions(depth_pyramid, 0)) * 0.5; // Halve the view-space AABB size as the depth pyramid is half the view size
+    let width = (aabb.z - aabb.x) * depth_pyramid_size_mip_0.x;
+    let height = (aabb.w - aabb.y) * depth_pyramid_size_mip_0.y;
+    let depth_level = max(0, i32(ceil(log2(max(width, height))))); // TODO: Naga doesn't like this being a u32
+    let depth_pyramid_size = vec2<f32>(textureDimensions(depth_pyramid, depth_level));
+    let aabb_top_left = vec2<u32>(aabb.xy * depth_pyramid_size);
+
+    let depth_quad_a = textureLoad(depth_pyramid, aabb_top_left, depth_level).x;
+    let depth_quad_b = textureLoad(depth_pyramid, aabb_top_left + vec2(1u, 0u), depth_level).x;
+    let depth_quad_c = textureLoad(depth_pyramid, aabb_top_left + vec2(0u, 1u), depth_level).x;
+    let depth_quad_d = textureLoad(depth_pyramid, aabb_top_left + vec2(1u, 1u), depth_level).x;
+    let occluder_depth = min(min(depth_quad_a, depth_quad_b), min(depth_quad_c, depth_quad_d));
+
+    // Check whether or not the cluster would be occluded if drawn
+    var meshlet_visible: bool;
+    if view.projection[3][3] == 1.0 {
+        // Orthographic
+        let sphere_depth = view.projection[3][2] + (culling_bounding_sphere_center_view_space.z + culling_bounding_sphere_radius) * view.projection[2][2];
+        meshlet_visible = sphere_depth >= occluder_depth;
+    } else {
+        // Perspective
+        let sphere_depth = -view.projection[3][2] / (culling_bounding_sphere_center_view_space.z + culling_bounding_sphere_radius);
+        meshlet_visible = sphere_depth >= occluder_depth;
     }
 
-#ifdef MESHLET_SECOND_CULLING_PASS
-    // In the second culling pass, cull against the depth pyramid generated from the first pass
+    // Write if the cluster should be occlusion tested in the second pass
+#ifdef MESHLET_FIRST_CULLING_PASS
```

Comment:
```suggestion
#ifdef MESHLET_FIRST_CULLING_PASS
    // Write if the cluster should be occlusion tested in the second pass
```

Reply:
It's easier to read if we keep the comments on top of the ifdefs imo, I'd rather not change it unless you feel strongly about it (but I also want to block this PR anymore than I have to :P)